### PR TITLE
Add row unique identifiers to metadata

### PR DESCRIPTION
**Pull request by jbwong05**
## Description
Adds the ability to add unique identifiers to rows via [metadata](https://developers.google.com/sheets/api/guides/metadata) as part of https://3.basecamp.com/4474129/buckets/38736474/todos/772966216. Metadata for google sheets allows for a key and a value to be associated with each row independently (the key and value are not tied to each other like they are in a traditional map).
When the scanning of the Onestop rows occurs, a Metadata object is created for each valid event row by setting the key (this causes Google Sheets to create the Metadata object). The value of the Metadata object contains the unique id of the Row. If it is `null`, this indicates it's unique has not yet been assigned indicating the row is new. These unique ids are tied to the specific row and will stay with the row even if the row is moved around in the spreadsheet tab.

## Changes
* Re-added `ts-loader` dependency that was removed as part of #3 as it is required to compile TS with webpack
* Added functions that allow checking, assigning, and retrieving unique ids for rows via metadata
* Adds a metadata field to the Row interface
* Removes a function with an output parameter which we are trying to avoid (added as part of #2 )

### CODE REVIEW

File: src/main/metadata.ts
Comment (andrewc5716) [inline src/main/metadata.ts:30]: non-blocking comment: in my mind "id" is usually unique, so i'm wondering if dropping "unique" from these function names would help with conciseness. not 100% sure of this suggestion but if we were to also infer we're working with rows b/c of the input argument, we could even call them "getId", "generateId", "hasId" etc.

File: src/main/scan.ts
Comment (andrewc5716) [inline src/main/scan.ts:75]: sick lol

**Review by andrewc5716 — DISMISSED** [review]

File: src/main/metadata.ts
Comment (andrewc5716) [inline src/main/metadata.ts:30]: as far as possible, we should try to avoid returning null - it's just waiting for a NPE down the road. i'd rather throw an exception immediately here and we can identify the source of the issue in a controlled manner, vs the NPE popping up somewhere else and it may take longer to debug why

Comment (andrewc5716) [reply src/main/metadata.ts:30]: i see it's used for rowHasUniqueId, but that could just use row.metadata.getValue() !== null it seems.
i don't want to block this PR since i'll be gone over the weekend, so you can feel free to address this in the next PR or get someone else's approval

Comment (jbwong05) [reply src/main/metadata.ts:30]: Will drop Row and Unique from the function names

File: src/main/scan.ts
Comment (jbwong05) [reply src/main/scan.ts:75]: lol

File: src/main/metadata.ts
Comment (jbwong05) [reply src/main/metadata.ts:30]: I'll address this in this PR by throwing an exception. Will get a review from someone else

File: src/main/row.ts
Comment (andrewc5716) [inline src/main/row.ts:68]: this makes me think for the future it might be helpful to add the sheet tab name and maybe the row number. out of scope for this pr, but we might want to consider adding it later when it makes sense

File: src/main/row.ts
Comment (andrewc5716) [inline src/main/row.ts:46]: nitpick, "assignId" makes me think we're going to pass in an id to assign. i'm wondering if something like "generateId" would be more clear, or if we even need to go more explicit like "generateIdForRow()"

Comment (jbwong05) [reply src/main/row.ts]: Renamed to `generateIdForRow()`

**Review by andrewc5716 — APPROVED** [review]

**Review by jkzang — APPROVED** [review]

**Pull request merged**
